File: src/main/java/io/codeka/gaia/repository/TerraformModuleRepository.java
package io.codeka.gaia.repository;

import io.codeka.gaia.bo.TerraformModule;
import io.codeka.gaia.teams.bo.Team;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TerraformModuleRepository extends MongoRepository<TerraformModule, String> {

    int countByAuthorizedTeamsContaining(Team team);

    List<TerraformModule> findAllByAuthorizedTeamsContaining(Team team);

    Optional<TerraformModule> findByIdAndAuthorizedTeamsContaining(String id, Team team);
}

File: src/main/java/io/gaia_app/modules/repository/TerraformModuleRepository.kt
package io.gaia_app.modules.repository

import io.gaia_app.modules.bo.TerraformModule
import io.gaia_app.organizations.Organization
import io.gaia_app.organizations.User
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface TerraformModuleRepository : MongoRepository<TerraformModule, String> {
    fun countByModuleMetadataCreatedBy(user: User): Long
    fun countByAuthorizedOrganizationsContainingOrModuleMetadataCreatedBy(organization: Organization, user: User): Long
    fun findAllByModuleMetadataCreatedByOrAuthorizedOrganizationsContaining(user: User, organization: Organization): List<TerraformModule>
    fun findAllByModuleMetadataCreatedBy(user: User): List<TerraformModule>
}
